From ffdf27b84904d4c4e8294ce22e5fd9c423cf0d7c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Marchesin?= <marcheu@chromium.org>
Date: Fri, 28 Apr 2017 11:56:59 -0700
Subject: [PATCH 16/39] CHROMIUM: i965: disable hiz on braswell

Hiz causes GPU hangs on braswell, so let's disable it.

BUG=b/35570762, b/35574152
TEST=run graphics_GLBench on 3 * kefka for a total of 45 hours, no GPU hangs observed

Change-Id: I57402696fb0e970f0a38d87a33f2179b294a2cf1
---
 src/mesa/drivers/dri/i965/brw_context.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/mesa/drivers/dri/i965/brw_context.c b/src/mesa/drivers/dri/i965/brw_context.c
index 50eb256..a73050b 100644
--- a/src/mesa/drivers/dri/i965/brw_context.c
+++ b/src/mesa/drivers/dri/i965/brw_context.c
@@ -846,7 +846,9 @@ brwCreateContext(gl_api api,
    brw->is_cherryview = devinfo->is_cherryview;
    brw->is_broxton = devinfo->is_broxton || devinfo->is_geminilake;
    brw->has_llc = devinfo->has_llc;
-   brw->has_hiz = devinfo->has_hiz_and_separate_stencil;
+   /* Braswell has hiz issues, disable it. */
+   brw->has_hiz = devinfo->has_hiz_and_separate_stencil
+			&& screen->deviceID != 0x22B1;
    brw->has_separate_stencil = devinfo->has_hiz_and_separate_stencil;
    brw->has_pln = devinfo->has_pln;
    brw->has_compr4 = devinfo->has_compr4;
-- 
2.7.4

